socketio_middleware/src/socketio.rs

```diff
@@ -438,11 +455,18 @@ impl SocketIOWrapper {
                     WSSocketMessage::WsPong => {
                         let _ = self.socket.send(Message::Ping([].to_vec())).await;
                     }
+
                     WSSocketMessage::Close => {
+                        // remove the socket from all joined rooms
+                        for room in &self.rooms {
```

Comment:
This is great! Let's actually move this whole for loop into the `close` fn. That way, if anyone else ever calls `close` it will leave all of the associated rooms as well!

Reply:
Yes, this should move to SocketIOWrapper::close()